python/demos/picture of the day viewer/app.py

```diff
@@ -0,0 +1,108 @@
+from flask import Flask, render_template, request, jsonify, redirect
+from datetime import date, timedelta
+import requests
+
+APP = Flask(__name__)
+SESSION = requests.Session()
+ENDPOINT = "https://en.wikipedia.org/w/api.php"
+
+current_date = date.today()
+
+@APP.route("/change_date", methods = ["POST"])
+def change_date():
+    global current_date
+
+    try:
+        change_date = request.form["change_date"]
+
+        if (change_date == "← Back"):
+            new_date = decrement_date()
+        elif (change_date == "Next →"):
+                new_date = increment_date()
+    except:
+        new_date = date.today()
+
+    current_date = new_date
+
+    return redirect("/")
+
+@APP.route("/", methods = ["GET", "POST"])
+
+def index():
+    if (request.method == "POST"):
+        date_to_fetch = str(current_date)
+        results = fetch_potd(date_to_fetch)
+
+        return jsonify(results = results)
+
+    return render_template("index.html")
+
+def increment_date():
+    return current_date + timedelta(days = 1)
+
+def decrement_date():
+    return current_date - timedelta(days = 1)
+
+def fetch_potd(date):
+    params = {
+        "action": "query",
+        "format": "json",
+        "formatversion": "2",
+        "prop": "images",
+        "titles": "Template:POTD protected/" + date
+    }
+
+    response = SESSION.get(url = ENDPOINT, params = params)
+    data = response.json()
+    try:
+        file_name = data["query"]["pages"][0]["images"][0]["title"]
+    
+        results = [{
+            "title": file_name,
+            "image": fetch_image_url(file_name),
+            "description": fetch_description(params["titles"]),
+            "date": date
+        }]
+    except:
+        results = [{
+            "title": "No Image For This Date",
+            "image": "https://upload.wikimedia.org/wikipedia/commons/4/49/404_Not_Found.png",
+            "description": "Wikipedia does not currently have an image associated with this date.",
+            "date": date
+        }]
+
+    return results
+
+def fetch_image_url(file_name):
```

Comment:
I'm wondering how we can remove these two additional calls to fetch URL and description. I cannot myself figure out a way to pull title, description, and URL with one call, but might be we could ask on Discourse if this is even possible?

Reply:
I removed one of them, the one that fetches the description, because the results weren't very high quality and it mostly just slowed loading down.